globus_sdk/transfer/paging.py

```diff
@@ -163,22 +179,68 @@ def iterable_func(self):
 
         This method is the real workhorse of this entire module.
         """
-        # now, cap the limit per request to the max per request size
-        limit = min(self.num_results, self.max_results_per_call)
+        if not self.client_kwargs['params']:
+            self.client_kwargs['params'] = {}
+
+        # to start with, cap the limit per request to the max per request size
+        self.limit = min(self.num_results, self.max_results_per_call)
+
+        def _set_params_for_next_call():
+            # if we're about to request more results than the user asked
+            # for, limit ourselves on the last paginated call to the API
+            if self.offset + self.limit > self.num_results:
+                self.limit = self.num_results - self.offset
+
+            # if the paging is done by marker, just carry over the marker
+            if self.paging_style == self.PAGING_STYLE_MARKER:
+                if self.marker:
+                    self.client_kwargs['params']['marker'] = self.marker
+            # these params work for all paging styles *except* MARKER
+            else:
+                self.client_kwargs['params']['offset'] = self.offset
+                self.client_kwargs['params']['limit'] = self.limit
+
+        def _check_has_next_page(res):
+            """
+            Check that the API says there are more results available.
+
+            Additionally, update the PaginatedResource.maker or
+            PaginatedResource.offset based on the response
+            """
+            # if the paging style is MARKER, look at the marker
+            if self.paging_style == self.PAGING_STYLE_MARKER:
+                # marker may be 0, null, or absent if no more results
+                # API docs aren't 100% clear -- looks like 0 is what we should
+                # expect, but we'll also accept null or absent to be safe
```

Comment:
I believe how this works (when I was playing around with this I looked at the Transfer code) is that `marker` is always an opaque value that marks where in set of available data you are (by default it will be `0` if you have all the results in this response, or if this is the beginning of the set of results). It shouldn't ever be `null` but there's no harm in handling that just to be safe.

If there are more pages of results, that will be indicated by the `next_marker` property in the response, which will be set to a value that the caller should use for the next requests. `next_marker` will be `null` if there are no more results available.

So basically, you make your first call as `/task/:id/successful_transfers` and you'll get back a payload that will have a `marker` set to some value that you could send back with `?marker=<val>` if you want to re-fetch the exact same set of results again. If the task has no more results, `next_marker` will be `null`. If there are more results, `next_marker` will be an opaque value that you can fetch with `?marker=<next_marker>`.